### ISSUE TITLE
A crashed bitcoin core client breaks eclair

### ISSUE THREAD

**Issue by dscotese** [ACINQ/eclair#935]
I opened the "Create a channel" dialog in eclair.  My instance of Bitcoin Core had crashed.  I restarted it without doing anything else with eclair.  It was four days behind.  When I stopped it, eclair popped up  a message saying it lost connection (of course!).  After the new instance was up for about 20 seconds, the "lost connection" message disappeared. About 5 minutes before the new instance of Bitcoin Core was completely up to date, Windows 10 (my OS - and i don't know my JDK version - how do you get that?) popped up a message saying that eclair had stopped responding.  I waited out the five minutes, expecting that when BTCCore was up to date, it would answer eclair and eclair would no longer be hanging.

But even a couple minutes after BTCCore was running and fully synced, eclair was still disabled and unresponsive.  I finally killed it and restarted it.  

I have not tried to reproduce the issue.  This was in version v0.2-beta9 downloaded from https://github.com/ACINQ/eclair/releases/download/v0.2-beta9/Eclair-0.2-beta9.exe

The logs I have for the days that BTCCore had to catch up show the following exception repeatedly (thousands of times):
```
2019-04-03 03:48:02,828 ERROR f.a.e.b.b.r.BatchingClient  - got exception for batch of 1 requestsjava.util.NoSuchElementException: Status code 500: Work queue depth exceeded
	at com.softwaremill.sttp.Response.unsafeBody(Response.scala:54)
	at fr.acinq.eclair.blockchain.bitcoind.rpc.BasicBitcoinJsonRPCClient$$anonfun$invoke$2.apply(BasicBitcoinJsonRPCClient.scala:49)
	at fr.acinq.eclair.blockchain.bitcoind.rpc.BasicBitcoinJsonRPCClient$$anonfun$invoke$2.apply(BasicBitcoinJsonRPCClient.scala:43)
	at scala.util.Success$$anonfun$map$1.apply(Try.scala:237)
	at scala.util.Try$.apply(Try.scala:192)
	at scala.util.Success.map(Try.scala:237)
	at scala.concurrent.Future$$anonfun$map$1.apply(Future.scala:237)
	at scala.concurrent.Future$$anonfun$map$1.apply(Future.scala:237)
	at scala.concurrent.impl.CallbackRunnable.run(Promise.scala:36)
	at scala.concurrent.impl.ExecutionContextImpl$AdaptedForkJoinTask.exec(ExecutionContextImpl.scala:121)
	at scala.concurrent.forkjoin.ForkJoinTask.doExec(ForkJoinTask.java:260)
	at scala.concurrent.forkjoin.ForkJoinPool$WorkQueue.pollAndExecAll(ForkJoinPool.java:1253)
	at scala.concurrent.forkjoin.ForkJoinPool$WorkQueue.runTask(ForkJoinPool.java:1346)
	at scala.concurrent.forkjoin.ForkJoinPool.runWorker(ForkJoinPool.java:1979)
	at scala.concurrent.forkjoin.ForkJoinWorkerThread.run(ForkJoinWorkerThread.java:107)
```
The first occurrence of that error in the long string of it happening repeatedly (several times a minute) was preceded by this:
```
2019-04-01 13:26:44,461 INFO  f.a.eclair.router.Router  n:02362eb8e582a890c7febe9293e3f085094ecdb66d29f9a9ee0447f12214f299e0 - validating shortChannelId=533224x614x1
2019-04-01 13:26:56,802 ERROR f.a.e.b.b.r.BatchingClient  - got exception for batch of 1 requestsjava.net.SocketTimeoutException: Read timed out
	at java.net.SocketInputStream.socketRead0(Native Method)
	at java.net.SocketInputStream.socketRead(SocketInputStream.java:116)
	at java.net.SocketInputStream.read(SocketInputStream.java:171)
	at java.net.SocketInputStream.read(SocketInputStream.java:141)
	at okio.Okio$2.read(Okio.java:140)
	at okio.AsyncTimeout$2.read(AsyncTimeout.java:237)
	at okio.RealBufferedSource.indexOf(RealBufferedSource.java:355)
	at okio.RealBufferedSource.readUtf8LineStrict(RealBufferedSource.java:227)
	at okhttp3.internal.http1.Http1Codec.readHeaderLine(Http1Codec.java:215)
	at okhttp3.internal.http1.Http1Codec.readResponseHeaders(Http1Codec.java:189)
	at okhttp3.internal.http.CallServerInterceptor.intercept(CallServerInterceptor.java:88)
	at okhttp3.internal.http.RealInterceptorChain.proceed(RealInterceptorChain.java:147)
	at okhttp3.internal.connection.ConnectInterceptor.intercept(ConnectInterceptor.java:45)
	at okhttp3.internal.http.RealInterceptorChain.proceed(RealInterceptorChain.java:147)
	at okhttp3.internal.http.RealInterceptorChain.proceed(RealInterceptorChain.java:121)
	at okhttp3.internal.cache.CacheInterceptor.intercept(CacheInterceptor.java:93)
	at okhttp3.internal.http.RealInterceptorChain.proceed(RealInterceptorChain.java:147)
	at okhttp3.internal.http.RealInterceptorChain.proceed(RealInterceptorChain.java:121)
	at okhttp3.internal.http.BridgeInterceptor.intercept(BridgeInterceptor.java:93)
	at okhttp3.internal.http.RealInterceptorChain.proceed(RealInterceptorChain.java:147)
	at okhttp3.internal.http.RetryAndFollowUpInterceptor.intercept(RetryAndFollowUpInterceptor.java:126)
	at okhttp3.internal.http.RealInterceptorChain.proceed(RealInterceptorChain.java:147)
	at okhttp3.internal.http.RealInterceptorChain.proceed(RealInterceptorChain.java:121)
	at okhttp3.RealCall.getResponseWithInterceptorChain(RealCall.java:200)

2019-04-01 13:26:56,802 INFO  f.a.eclair.router.Router  - got validation result for shortChannelId=563887x1905x0 (awaiting=20 stash.nodes=4 stash.updates=37)
2019-04-01 13:26:56,802 WARN  f.a.eclair.router.Router  - validation failure for shortChannelId=563887x1905x0 reason=Read timed out
```

**Comment by sstone** [comment]
To work properly eclair needs to be connected to a synchronised bitcoin node. Restarting bitcoin core without restarting eclair will work if you just need to change a few configuration parameters for example, but if it has been stopped for 4 days then it won't respond to queries while it's syncing and it will have side-effects on eclair and which are out-of-scope atm. 
TL;DR if you bitcoin node has been stopped for a few days, stop eclair, wait until your bitcoin node is synced, then start eclair.
